lib/connection.js

```diff
@@ -227,7 +227,8 @@ C.open = function(allFields, openCallback0) {
       tunedOptions.channelMax =
         negotiate(fields.channelMax, allFields.channelMax);
       tunedOptions.heartbeat =
-        negotiate(fields.heartbeat, allFields.heartbeat);
+        allFields.heartbeat === 0 ?
```

Comment:
If this isn't set by the user, it should accept the server value. However, it's defaulted to `0` in connect.js, so code that is currently not supplying a value will now disable heartbeats rather than using the server value.